dds/DCPS/DiscoveryBase.h

```diff
@@ -1051,10 +1233,57 @@ namespace OpenDDS {
           subQos = &tempSubQos;
           cfProp = &dsi->second.reader_data_.contentFilterProperty;
           reader_type_info = &dsi->second.type_info_;
+          topic_name = dsi->second.get_topic_name();
         } else {
           return; // Possible and ok, since lock is released
         }
 
+        // check consistency
+        bool consistent = false;
+
+        typename OPENDDS_MAP(OPENDDS_STRING, TopicDetails)::iterator td_iter = topics_.find(topic_name);
+        if (td_iter == topics_.end()) {
+          if (DCPS::DCPS_debug_level) {
+            ACE_DEBUG((LM_ERROR,
```

Comment:
```suggestion
            ACE_ERROR((LM_ERROR,
```

Reply:
If it is an error it shouldn't be behind the debug level should it?

Reply:
Good question. If the error is going to be directly returned to the user through the public API it should be at debug level 1. That is obviously not the case here though, so yea the error log should be unconditional. Maybe also a little more information? Like GUID(s) and the topic name?